pkg/controller/prune.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"fmt"
+
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/golang/glog"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/brownfield"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/events"
+)
+
+type pruneFunc func(c *AppGwIngressController, appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingressList []*v1beta1.Ingress) []*v1beta1.Ingress
+
+var pruneFuncList = []pruneFunc{pruneNoPrivateIP, pruneProhibitedIngress}
```

Comment:
Does `pruneNoPrivateIP` need to be brownfield-aware?
It seems that the answer is - no. A prohibited target would get removed at some point, so even if I have a VMSS-hosted app that uses the same App Gateway and a PublicIP Listener - as long as it is prohibited - it won't matter.

It might still be good to:
1. order these functions
2. leave a comment explaining why it is important to order these

If the order of functions perhaps should be:
  1. `pruneProhibitedIngress`
  2. `pruneNoPrivateIP`

I'm imagining a situation where I have prohibited a target, where I have a VMSS-hosted app with a Public IP.

Where we log pruned items from `pruneNoPrivateIP` it would potentially emit messages about prohibited targets being pruned for having a Public IP.